server/src/main/java/io/unitycatalog/server/service/IcebergRestCatalogService.java

```diff
@@ -149,7 +149,7 @@ public HttpResponse tableExists(@Param("namespace") String namespace,
                                   @Param("table") String table) {
     List<String> namespaceParts = Splitter.on(".").splitToList(namespace);
     if (namespaceParts.size() != 2) {
-      throw new IllegalArgumentException("invalid namespace " + namespace);
+      throw new BaseException(ErrorCode.INVALID_ARGUMENT, "invalid namespace " + namespace);
     }
```

Comment:
Nit: suggest refactoring this repeated code into a separate method like the following:

```
  private List<String> splitTwoPartNamespace(String namespace) {
    List<String> namespaceParts = Splitter.on(".").splitToList(namespace);
    if (namespaceParts.size() != 2) {
      String errMsg = "Invalid two-part namespace " + namespace;
      throw new BaseException(ErrorCode.INVALID_ARGUMENT, errMsg);
    }

    return namespaceParts;
  }
```

Reply:
Good suggestion! will update

Reply:
Good suggestion! Will update